graylog2-server/src/main/java/org/graylog2/cluster/lock/MongoLockService.java

```diff
@@ -54,20 +57,41 @@
  */
 @Singleton
 public class MongoLockService implements LockService {
-    private static final Duration LOCK_TTL = Duration.ofSeconds(60);
 
-    private static final String COLLECTION_NAME = "cluster_locks";
+    public static final String COLLECTION_NAME = "cluster_locks";
+    public static final java.time.Duration MIN_LOCK_TTL = Duration.ofSeconds(60);
 
     private final NodeId nodeId;
     private final MongoCollection<Document> collection;
 
     @Inject
-    public MongoLockService(NodeId nodeId, MongoConnection mongoConnection) {
+    public MongoLockService(NodeId nodeId,
+                            MongoConnection mongoConnection,
+                            @Named("leader_election_lock_ttl") Duration leaderElectionLockTTL) {
         this.nodeId = nodeId;
 
         collection = mongoConnection.getMongoDatabase().getCollection(COLLECTION_NAME);
+
         collection.createIndex(Indexes.ascending(FIELD_RESOURCE_NAME), new IndexOptions().unique(true));
-        collection.createIndex(Indexes.ascending(FIELD_UPDATED_AT), new IndexOptions().expireAfter(LOCK_TTL.getSeconds(), TimeUnit.SECONDS));
+
+        final Bson updatedAtKey = Indexes.ascending(FIELD_UPDATED_AT);
+        final IndexOptions indexOptions = new IndexOptions().expireAfter(leaderElectionLockTTL.getSeconds(), TimeUnit.SECONDS);
+        ensureTTLIndex(collection, updatedAtKey, indexOptions);
+    }
+
+    // MongoDB Indexes cannot be altered once created. If the leaderElectionLockTTL has changed, replace the index
+    private void ensureTTLIndex(MongoCollection<Document> collection, Bson updatedAtKey, IndexOptions indexOptions) {
+        for (Document document : collection.listIndexes()) {
+            final Set<String> keySet = document.get("key", Document.class).keySet();
+            if (keySet.contains(FIELD_UPDATED_AT)) {
+                if (Objects.equals(document.getLong("expireAfterSeconds"), indexOptions.getExpireAfter(TimeUnit.SECONDS))) {
```

Comment:
With MongoDB 5.0 this is broken :
```
Exception in thread "main" com.google.inject.CreationException: Unable to create injector, see the following errors:

1) [Guice/ErrorInjectingConstructor]: ClassCastException: Integer cannot be cast to Long
  at MongoLockService.<init>(MongoLockService.java:70)
  at MongoLockService.class(MongoLockService.java:62)
  while locating MongoLockService
  at LeaderElectionModule.configure(LeaderElectionModule.java:43)
      \_ installed by: ServerBindings -> LeaderElectionModule
  at AutomaticLeaderElectionService.<init>(AutomaticLeaderElectionService.java:50)
      \_ for 2nd parameter
  at AutomaticLeaderElectionService.class(AutomaticLeaderElectionService.java:34)
  while locating AutomaticLeaderElectionService
  at LeaderElectionModule.configure(LeaderElectionModule.java:41)
      \_ installed by: ServerBindings -> LeaderElectionModule
  at PersistedInputsImpl.<init>(PersistedInputsImpl.java:45)
      \_ for 3rd parameter
  while locating PersistedInputsImpl
  at InputLauncher.<init>(InputLauncher.java:47)
      \_ for 3rd parameter
  at InputSetupService.<init>(InputSetupService.java:50)
      \_ for 3rd parameter
  at InputSetupService.class(InputSetupService.java:40)
  while locating InputSetupService
  while locating Service annotated with @Element(setName=,uniqueId=11, type=MULTIBINDER, keyType=)
  at ServiceManagerProvider.services(ServiceManagerProvider.java:34)
      \_ for field services
  while locating ServiceManagerProvider
  at GenericBindings.configure(GenericBindings.java:81)
  while locating ServiceManager

Learn more:
  https://github.com/google/guice/wiki/ERROR_INJECTING_CONSTRUCTOR
Caused by: ClassCastException: Integer cannot be cast to Long
	at Document.getLong(Document.java:266)
	at MongoLockService.ensureTTLIndex(MongoLockService.java:87)
	at MongoLockService.<init>(MongoLockService.java:79)
	at MongoLockService$$FastClassByGuice$$213723370.GUICE$TRAMPOLINE(<generated>)
	at MongoLockService$$FastClassByGuice$$213723370.apply(<generated>)
	at DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:82)
	at ConstructorInjector.provision(ConstructorInjector.java:114)
	at ConstructorInjector.construct(ConstructorInjector.java:91)
	at ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:296)
	at ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at SingletonScope$1.get(SingletonScope.java:169)
	at InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45)
	at FactoryProxy.get(FactoryProxy.java:60)
	at ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at SingletonScope$1.get(SingletonScope.java:169)
	at InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45)
	at SingleParameterInjector.inject(SingleParameterInjector.java:40)
	at SingleParameterInjector.getAll(SingleParameterInjector.java:60)
	at ConstructorInjector.provision(ConstructorInjector.java:113)
	at ConstructorInjector.construct(ConstructorInjector.java:91)
	at ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:296)
	at ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at SingletonScope$1.get(SingletonScope.java:169)
	at InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45)
	at FactoryProxy.get(FactoryProxy.java:60)
	at ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at SingletonScope$1.get(SingletonScope.java:169)
	at InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45)
	at SingleParameterInjector.inject(SingleParameterInjector.java:40)
	at SingleParameterInjector.getAll(SingleParameterInjector.java:60)
	at ConstructorInjector.provision(ConstructorInjector.java:113)
	at ConstructorInjector.construct(ConstructorInjector.java:91)
	at ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:296)
	at FactoryProxy.get(FactoryProxy.java:60)
	at SingleParameterInjector.inject(SingleParameterInjector.java:40)
	at SingleParameterInjector.getAll(SingleParameterInjector.java:60)
	at ConstructorInjector.provision(ConstructorInjector.java:113)
	at ConstructorInjector.construct(ConstructorInjector.java:91)
	at ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:296)
	at SingleParameterInjector.inject(SingleParameterInjector.java:40)
	at SingleParameterInjector.getAll(SingleParameterInjector.java:60)
	at ConstructorInjector.provision(ConstructorInjector.java:113)
	at ConstructorInjector.construct(ConstructorInjector.java:91)
	at ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:296)
	at ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at SingletonScope$1.get(SingletonScope.java:169)
	at InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45)
	at FactoryProxy.get(FactoryProxy.java:60)
	at SingleParameterInjector.inject(SingleParameterInjector.java:40)
	at RealMultibinder$RealMultibinderProvider.doProvision(RealMultibinder.java:206)
	at RealMultibinder$RealMultibinderProvider.doProvision(RealMultibinder.java:159)
	at InternalProviderInstanceBindingImpl$Factory.get(InternalProviderInstanceBindingImpl.java:113)
	at SingleFieldInjector.inject(SingleFieldInjector.java:50)
	at MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:146)
	at ConstructorInjector.provision(ConstructorInjector.java:124)
	at ConstructorInjector.construct(ConstructorInjector.java:91)
	at ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:296)
	at BoundProviderFactory.get(BoundProviderFactory.java:58)
	at ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at SingletonScope$1.get(SingletonScope.java:169)
	at InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45)
	at InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:213)
	at InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:186)
	at InternalInjectorCreator.build(InternalInjectorCreator.java:113)
	at Guice.createInjector(Guice.java:87)
	at GuiceInjectorHolder.createInjector(GuiceInjectorHolder.java:34)
	at CmdLineTool.setupInjector(CmdLineTool.java:478)
	at CmdLineTool.run(CmdLineTool.java:283)
	at Main.main(Main.java:45)
```

Looks like we're getting the value back as an integer 😡